nucypher/characters/lawful.py

```diff
@@ -1748,34 +1735,29 @@ def interface_info_with_metadata(self):
     #
 
     def verify_kfrag_authorization(self,
-                                   alice: Alice,
-                                   kfrag: KeyFrag,
-                                   signed_writ: bytes,
-                                   work_order: 'WorkOrder'
+                                   hrac: bytes,
+                                   provider: Alice,
```

Comment:
Repeating my comment from discord here.

"""
I'm iffy on `provider` since it is less specific; there is the generator of Kfrags and the publisher of policies

perhaps something like `kfrag_creator`/`kfrag_generator`... 🤷 
"""

Reply:
"author" and "publisher" dynamic could be good

Reply:
Yeah "author and publisher" is pretty elegant.

Reply:
> "author" sounds exactly like Alice

To me, Alice is a specific instance (character, person, whatever), Author is a type (or a role). Like, we can have `alice = Author()` in some example code, meaning that we define a character Alice who is an Author. `Alice` as a type always felt to me like it should be a singleton or something.